indigo_social/views.py

```diff
@@ -96,6 +99,10 @@ def form_invalid(self, form):
         self.form = form
         return redirect(self.get_success_url())
 
+    def get_object(self, queryset=None):
+        self.kwargs['pk'] = UserProfile.objects.get(user=User.objects.get(username=self.kwargs['username'])).pk
```

Comment:
You can simply return the userprofile object here and not call super. Otherwise, you're going to the db to get the user, then the userprofile, then getting the pk, then Django goes back to the db to get the userprofile object again.

Reply:
for this I just also changed the view to use the User model, so if I've understood correctly the above doesn't apply?